### PR TITLE
Adds the Leicester Repeater

### PR DESCRIPTION
**Pull request by joniroq**
## About The Pull Request

Adds the Leicester Repeater to the marine vendor (Rifle Category). Using .44 rounds it can hold 14 rounds plus one in the chamber. It fires considerably slower than the T-44 Revolver (comparable to the T-39 in terms of speed) and so DPS wise it is actually more effective to use two revolvers. The only real advantage is the lower damage fall off allowing it to be used more like a rifle at range.

Currently it doesn't animate. Will look to fix that at a later date.

## Why It's Good For The Game

Rootin' Tootin' Beno Shootin'. Only Lever action gun we have is the Mare's Leg, which is terrible. This adds an option for players that want to complete the cowboy loadout (sans cowboy hat)

## Changelog
:cl: Joniroq
add: Added Leicester Repeater. Uses .44 caliber rounds. Yeehaw!

/:cl:

### CODE REVIEW

**Comment by joniroq** [comment]
Wait where the hell did the sprites go?

**Comment by Pariah919** [comment]
disagree with making an entirely new gun with this, replace the mare's leg, considering this is basically a clone of it. And would ask to relent from refrencing a specific model of lever action rifle, make up a name or something.

As for balancing, set up 1 second FD and 0.2 Pump delay and give it a 1.5x damage mult.

**Review by BraveMole — CHANGES_REQUESTED** [review]
I don't really care about it being a duplicate of a surv weapon, but do the rest of what MJP said please.

**Comment by joniroq** [comment]
Drafting this for now as I won't have time to work on it til monday or tuesday. @BraveMole

File: code/modules/projectiles/guns/shotguns.dm [hunk 1]
```diff
@@ -808,6 +808,66 @@ can cause issues with ammo types getting mixed up during the burst.
 /obj/item/weapon/gun/shotgun/pump/lever/pump_notice(mob/user)
 	to_chat(user, span_notice("<b>You work [src] lever.</b>"))
 
+/obj/item/weapon/gun/shotgun/pump/lever/unload(mob/user)
```
Comment (BraveMole) [inline code/modules/projectiles/guns/shotguns.dm:811]: you messed up and this is duplicate

File: code/modules/projectiles/guns/shotguns.dm [hunk 2]
```diff
@@ -808,6 +808,66 @@ can cause issues with ammo types getting mixed up during the burst.
...
+	recoil_unwielded = 2
+	pump_delay = 2
+	aim_slowdown = 0.6
+
+/obj/item/weapon/gun/shotgun/pump/lever/pump_fail_notice(mob/user)
+	playsound(user,'sound/weapons/throwtap.ogg', 25, 1)
+	to_chat(user,span_warning("<b>[src] lever has already been worked, locking the lever; fire or unload a round to unlock it.</b>"))
+	recent_notice = world.time
+
+/obj/item/weapon/gun/shotgun/pump/lever/pump_notice(mob/user)
```
Comment (BraveMole) [inline code/modules/projectiles/guns/shotguns.dm:868]: duplicate

File: code/modules/projectiles/guns/shotguns.dm [hunk 3]
```diff
@@ -808,6 +808,66 @@ can cause issues with ammo types getting mixed up during the burst.
...
+	damage_mult = 1.5
+	damage_falloff_mult = 0.5
+	scatter = -5
+	scatter_unwielded = 15
+	recoil = 0
+	recoil_unwielded = 2
+	pump_delay = 2
+	aim_slowdown = 0.6
+
+/obj/item/weapon/gun/shotgun/pump/lever/pump_fail_notice(mob/user)
```
Comment (BraveMole) [inline code/modules/projectiles/guns/shotguns.dm:863]: duplicate

File: code/modules/projectiles/guns/shotguns.dm [hunk 4]
```diff
@@ -808,6 +808,66 @@ can cause issues with ammo types getting mixed up during the burst.
...
 
+/obj/item/weapon/gun/shotgun/pump/lever/unload(mob/user)
+	if(pump_lock)
+		to_chat(user, span_notice("<b>You pull [src]'s lever downward, ejecting the cartridge.</b>"))
+		pump_lock = FALSE //we're operating the slide release to unload, thus unlocking the pump
+	return ..()
+// ***********************************************
+// Winchester Rifle. The gun that won the west.
+
+/obj/item/weapon/gun/shotgun/pump/repeater
```
Comment (BraveMole) [inline code/modules/projectiles/guns/shotguns.dm:819]: you've put this in the middle of the lever proc, put that in the end of the file for exemple

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/shotguns.dm (hunk 3 above)
Comment (joniroq) [reply code/modules/projectiles/guns/shotguns.dm:863]: That's the gun?

Comment (BraveMole) [reply code/modules/projectiles/guns/shotguns.dm:863]: That's not the gun read closely

Comment (joniroq) [reply code/modules/projectiles/guns/shotguns.dm]: I think I understand what you mean now.

**Review by BraveMole — APPROVED** [review]

**Review by Kuro020 — CHANGES_REQUESTED** [review]
the obj sprite is good but the mob sprites are just the mosin,guns should have unique mob sprites even if they're a bit scuffed

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 5]
```diff
@@ -92,6 +92,7 @@
 			/obj/item/ammo_magazine/rifle/standard_skirmishrifle = -1,
 			/obj/item/weapon/gun/rifle/tx11 = -1,
 			/obj/item/ammo_magazine/rifle/tx11 = -1,
+			/obj/item/weapon/gun/shotgun/pump/repeater = -1,
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/marine_vending.dm:95]:
```suggestion
			/obj/item/weapon/gun/shotgun/pump/lever/repeater = -1,
```

**Review by BraveMole — CHANGES_REQUESTED** [review]
image

this doesn't compile

**Review by BraveMole — APPROVED** [review]

**Pull request merged**